index.ts

```diff
@@ -66,28 +97,33 @@ async function sendGroupEvent(invoice, customer, due_last_month, due_total, paid
     })
 }
 
-function last_month(invoices, key) {
+function last_month(global, invoices: StoredInvoice[], key) {
     const today = new Date()
     const firstDayThisMonth = Math.floor(new Date(today.getFullYear(), today.getMonth(), 1) / 1000)
     const firstDayNextMonth = Math.floor(new Date(today.getFullYear(), today.getMonth() + 1, 1) / 1000)
     return invoices
-        .filter(({ period_end }) => {
-            return period_end > firstDayThisMonth && period_end < firstDayNextMonth
+        .filter((invoice) => {
+            const timestamp = global.getInvoiceTimestamp(invoice)
+            return (
+                timestamp !== undefined  // older invoices may not have all timestamp data
+                && timestamp > firstDayThisMonth
+                && timestamp < firstDayNextMonth
+            )
         })
         .map((invoice) => invoice[key])
         .reduce((prev, cur) => prev + cur, 0)
 }
 
 async function sendInvoiceEvent(invoice, customer, global, storage, groupAddition) {
-    const paid_last_month = last_month(customer.invoices, 'amount_paid')
+    const paid_last_month = last_month(global, customer.invoices, 'amount_paid')
     const paid_total = customer.invoices.reduce((prev, cur) => prev.amount_paid + cur.amount_paid, { amount_paid: 0 })
 
-    const due_last_month = last_month(customer.invoices, 'amount_paid')
+    const due_last_month = last_month(global, customer.invoices, 'amount_paid')
     const due_total = customer.invoices.reduce((prev, cur) => prev.amount_paid + cur.amount_paid, { amount_paid: 0 })
 
     posthog.capture('Stripe Invoice Paid', {
         distinct_id: customer?.distinct_id,
-        timestamp: toISOString(invoice.period_end),
+        timestamp: global.getInvoiceTimestamp(invoice).toISOString(),
```

Comment:
`invoice` here isn't actually a `StoredInvoice` per the function signature… but they are the same general shape. This _is_ a hack, but not sure introducing another type to distinguish between the unaltered API response and the slimmed down one is worth it?